packages/client/src/components/shared/IndividualMetricView.tsx

```diff
@@ -0,0 +1,250 @@
+import * as React from 'react';
+import {observer} from 'mobx-react';
+import './IndividualMetricView.scss';
+import {Line} from 'react-chartjs-2';
+import NumberFormat from 'react-number-format';
+import {
+  CanaryAnalysisResult,
+  MetricSetPair,
+  MetricSetPairAttributes
+} from '../../domain/Kayenta';
+import humanFormat from 'human-format';
+import {metricSourceIntegrations} from '../../metricSources';
+
+interface IndividualMetricViewProps {
+  selectedMetric: string;
+  metricSourceType: string;
+  canaryAnalysisResultByIdMap: KvMap<CanaryAnalysisResult>;
+  metricSetPairsByIdMap: KvMap<MetricSetPair>;
+}
+
+const ROUNDING_POSITION: number = 4;
+
+@observer
+export default class IndividualMetricView extends React.Component<IndividualMetricViewProps> {
+  render(): React.ReactNode {
+    const {selectedMetric, metricSourceType, canaryAnalysisResultByIdMap, metricSetPairsByIdMap} = this.props;
+    const baselineData = metricSetPairsByIdMap[selectedMetric].values.control;
+    const canaryData = metricSetPairsByIdMap[selectedMetric].values.experiment;
+    const {startTimeMillis, stepMillis} = metricSetPairsByIdMap[selectedMetric].scopes.control;
+
+    const queryMapper: (
```

Comment:
This can and will be null in some cases, I am not sure why you are telling TS that this wont be null with the `!` operator. This will crash the app in some cases.